src/bci_build/package/__init__.py

```diff
@@ -118,6 +118,51 @@ def __str__(self) -> str:
         return self.value
 
 
+@enum.unique
+class Registry(enum.Enum):
+    """Supported registries for container image distribution"""
+
+    SUSE_COM = "registry.suse.com"
+    APP_COLL = "dp.apps.rancher.io"
+    OPENSUSE_ORG = "registry.opensuse.org"
+
+    def __str__(self) -> str:
+        return self.value
+
+    @property
+    def application_prefix(self) -> str:
+        """Container image prefix for application containers."""
+        match self.value:
+            case Registry.SUSE_COM.value:
```

Comment:
this is what I personally dislike. tons of if/match statements that specialize behavior based on some magic attributes. we should delegate to a special object that has the right context rather than repeating in every property some match statement wars that doesn't make the code look elegant or easy to read.